x/psiphon/psiphon.go

```diff
@@ -199,6 +199,18 @@ func (d *Dialer) Stop() error {
 	return nil
 }
 
+// If the dialer has not yet started, starts it.
+// If the dialer is already started, stop it and restart with the new config
+// TODO: if the dialer is already started and config is identical, just return nil
+func (d *Dialer) StartOrRestart(startCtx context.Context, config *DialerConfig) error {
```

Comment:
Let's rename `Start` to `Restart` or `Reconfigure` and handle the stop there if it's already started.

We should never return `errAlreadyStarted` because the call is actually to reconfigure the dialer, we shouldn't just use the running config. We should stop it and start it again.

This should fix the smart dialer problem, because the Restart won't return an error enymore.